### PR TITLE
docs: revised the online-offline event detection feature page

### PR DESCRIPTION
**Pull request by bandantonio**
#### Description of Change
Revised the online-offline event detection feature page, reworked code samples, added screenshot that corresponds to the code sample

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Revised the online-offline event detection feature page, reworked code samples, added screenshot that corresponds to the code sample

### CODE REVIEW

**Review by zcbenz — CHANGES_REQUESTED** [review]
We are going to add an API to return online status in the main process, I think we can use the new API in this doc and merge after the API gets merged.

https://github.com/electron/electron/pull/21004

File: docs/tutorial/online-offline-events.md [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+Since many cases return `true`, you should treat with care situations of
+getting false positives, as we cannot always assume that `true` value means
+that Electron can access the Internet. For example, in cases when the computer
+is running a virtualization software that has virtual Ethernet adapters in "always
+connected" state. Therefore, if you want to determine the Internet access
+status of Electron, you should develop additional means for this check.
+
+## Example
+
+## Event detection in the Renderer process
```
Comment (erickzhao) [inline docs/tutorial/online-offline-events.md:24]:
```suggestion
### Event detection in the Renderer process
```

File: docs/tutorial/online-offline-events.md [hunk 2]
```diff
@@ -21,33 +38,39 @@ app.whenReady().then(() => {
...
-There may be instances where you want to respond to these events in the
-main process as well. The main process however does not have a
-`navigator` object and thus cannot detect these events directly. Using
-Electron's inter-process communication utilities, the events can be forwarded
-to the main process and handled as needed, as shown in the following example.
+After launching the Electron application, you should see the notification:
+
+![Online-offline-event detection](../images/online-event-detection.png)
+
+## Event detection in the Main process
```
Comment (erickzhao) [inline docs/tutorial/online-offline-events.md:63]:
```suggestion
### Event detection in the Main process
```

File: docs/tutorial/online-offline-events.md [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
-Example:
+[Online and offline event](https://developer.mozilla.org/en-US/docs/Online_and_offline_events)
+detection can be implemented in the Renderer process using the
+[`navigator.onLine`](http://html5index.org/Offline%20-%20NavigatorOnLine.html)
+attribute, part of standard HTML5 API.
 
-_main.js_
+The `navigator.onLine` attribute returns:
+* `false` if any network requests are guaranteed to fail, i.e., definitely
+offline (disconnected from the network).
```
Comment (erickzhao) [inline docs/tutorial/online-offline-events.md:12]:
```suggestion
The `navigator.onLine` attribute returns:
* `false` if all network requests are guaranteed to fail (e.g. when disconnected
from the network).
```

**Review by erickzhao — COMMENTED** [review]
Comments for change.

@zcbenz: does this mean that we should wait for the next Electron release to land in stable? I think `/docs/guides/` fetches its data from `master`, so the `net.online` API won't be usable for most users for a while after it lands.

**Comment by zcbenz** [comment]
> @zcbenz: does this mean that we should wait for the next Electron release to land in stable? I think `/docs/guides/` fetches its data from `master`, so the `net.online` API won't be usable for most users for a while after it lands.

Hmm I am fine landing current document first and then update it after the new API is stable, it would require some extra work though.

**Comment by erickzhao** [comment]
Sounds good to me. @bandantonio's Season of Docs term is only until December, so let's ship faster and revisit later.

**Review by erickzhao — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Pull request merged**